src/core/stripe/services/Stripe.ts

```diff
@@ -17,25 +17,15 @@ const stripe = (() => {
 const createCheckoutSession: (params: StripeCheckoutParams ) => Promise<string> = async (
     params
 ) => {
-    // Define payment method for one time payment
-    const paymenyMethod = 'payment';
-
     // Prepare checkout generic session parameters
     const sessionParam: Stripe.Checkout.SessionCreateParams = {
-        mode: paymenyMethod,
-        line_items: [
-          {
-            price: params.priceId,
-            quantity: 1,
-          },
-        ],
+        mode: params.paymentMethod,
+        invoice_creation: {
+            enabled: true,
+        },
+        line_items: params.lineItems,
         success_url: `${process.env.STRIPE_SUCCESS_URL}`,
         cancel_url: `${process.env.STRIPE_CANCEL_URL}`,
-        payment_intent_data: {
-            metadata: {
-                'survey_id': params.surveyId,
-            }
-        },
       }
 
     // Add customer to session if exists
```

Comment:
This logic is very specific for Alliansis, I think that rather that force to send a customerid or a email to create a new stripe customer we should find a way to parameterize this to let the user choose what they want to do.
A way to do it would be to validate that either customer or email are populated, if both come as parameter an error should be displayed.
If customer is not populated then you could add the customer_creation if sent by parameter as its totally optional.